Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverInformation.hs

```diff
@@ -208,22 +210,55 @@ findAllWithLimitOffsetByMerchantId mbSearchString mbSearchStrDBHash mbLimit mbOf
 
 getDriversWithOutdatedLocationsToMakeInactive :: Transactionable m => UTCTime -> m [Person]
 getDriversWithOutdatedLocationsToMakeInactive before = do
-  findAll $ do
-    (driverInformation :& _ :& person) <-
-      from $
-        table @DriverInformationT
-          `innerJoin` table @DriverLocationT
-            `Esq.on` ( \(driverInformation :& drLoc) ->
-                         driverInformation ^. DriverInformationDriverId ==. drLoc ^. DriverLocationDriverId
-                           &&. drLoc ^. DriverLocationUpdatedAt <. val before
-                     )
-          `innerJoin` table @PersonT
-            `Esq.on` ( \(driverInformation :& _ :& person) ->
-                         driverInformation ^. DriverInformationDriverId ==. person ^. PersonTId
-                     )
-    where_ $ driverInformation ^. DriverInformationActive
-    orderBy [asc $ driverInformation ^. DriverInformationUpdatedAt]
-    pure person
+  driverInfos <- getDriverInfos
+  driverLocations <- getDriverLocs driverInfos before
+  drivers <- getDrivers driverLocations
+  return drivers
+
+getDrivers ::
+  Transactionable m =>
+  [DriverLocation] ->
+  m [Person]
+getDrivers driverLocs = do
+  Esq.findAll $ do
+    persons <- from $ table @PersonT
+    where_ $
+      persons ^. PersonTId `in_` valList personsKeys
+    return persons
+  where
+    personsKeys = toKey . cast <$> fetchDriverIDsFromLocations driverLocs
+
+getDriverInfos ::
+  Transactionable m =>
+  m [DriverInformation]
+getDriverInfos = do
+  Esq.findAll $ do
+    driverInfos <- from $ table @DriverInformationT
+    where_ $
+      driverInfos ^. DriverInformationActive
+    orderBy [asc $ driverInfos ^. DriverInformationUpdatedAt]
```

Comment:
Is this `orderBy` required?
It adds additional cost to the query execution time

Reply:
as we are filtering based on `before` time, I think orderBy is not required. I have removed it, will confirm with @piyushKumar-1 if this was added for some reasons